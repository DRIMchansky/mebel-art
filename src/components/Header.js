// packages
import React from 'react'
import styled from 'styled-components'
// components
import Logo from '../components/Logo'
import Nav from '../components/Nav'
import Container from '@material-ui/core/Container'
// utils
import { colors, breakpoints } from '../util/cssConfig'

const Header = ({ active, setActive }) => {
  return (
    <HeaderStyled>
      <ContainerStyled disableGutters>
        <Logo desctoponly="true" />
        <Nav />
      </ContainerStyled>
    </HeaderStyled>
  )
}

const HeaderStyled = styled.header`
  position: absolute;
  left: 0;
  top: 0;
  z-index: 0;
  width: 100%;
  height: 100%;
  background: ${colors.dark};

  @media (min-width: ${breakpoints.small}) {
    position: relative;
    background: ${colors.white};
    padding: 0;
  }
`
const ContainerStyled = styled(Container)`
  @media (min-width: ${breakpoints.small}) {
    /* Overwrite MaterialUI */
    display: flex !important;
    padding: 0 0.5rem !important;
    justify-content: space-between;
  }
`

export default Header
