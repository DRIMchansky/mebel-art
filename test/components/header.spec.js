import React from 'react'
import { mount } from 'enzyme'
import renderer from 'react-test-renderer'
import Header from '../../src/components/header'

test('Without props', () => {
  const tree = renderer.create(<Header />).toJSON()
  expect(tree).toMatchSnapshot()
})
