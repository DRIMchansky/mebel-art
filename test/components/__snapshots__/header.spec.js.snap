// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  font-weight: 300;
  color: #1c5796;
  font-size: 1.3rem;
  -webkit-letter-spacing: -1px;
  -moz-letter-spacing: -1px;
  -ms-letter-spacing: -1px;
  letter-spacing: -1px;
  display: none;
  visibility: hidden;
}

.c2 span {
  font-weight: 800;
  margin-left: 2px;
  color: #283646;
}

.c3 {
  margin-top: 30px;
}

.c3 ul {
  list-style-type: none;
  margin: 0;
  padding: 0;
  max-width: 60%;
}

.c3 a {
  text-transform: uppercase;
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #fffffe;
  padding: 1rem 1.4rem;
  font-size: 0.8rem;
  -webkit-transition: opacity 0.1s ease;
  transition: opacity 0.1s ease;
}

.c3 a:hover,
.c3 a:focus {
  opacity: 0.8;
}

.c0 {
  position: absolute;
  left: 0;
  top: 0;
  z-index: -1;
  width: 100%;
  height: 100%;
  background: #1c5796;
}

@media (min-width:600px) {
  .c2 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    visibility: visible;
  }
}

@media (min-width:600px) {
  .c3 ul {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (min-width:600px) {
  .c3 a {
    color: #283646;
    padding: 1.5rem 1rem;
  }
}

@media (min-width:600px) {
  .c3 {
    margin: 0;
  }
}

@media (min-width:600px) {
  .c0 {
    z-index: 0;
    position: relative;
    background: #fffffe;
    padding: 0;
  }
}

@media (min-width:600px) {
  .c1 {
    display: -webkit-box !important;
    display: -webkit-flex !important;
    display: -ms-flexbox !important;
    display: flex !important;
    padding: 0 1rem !important;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

<header
  className="c0"
>
  <div
    className="MuiContainer-root c1 MuiContainer-disableGutters MuiContainer-maxWidthXl"
  >
    <a
      className="c2"
      desctoponly="true"
      href="/"
    >
      Мебель
      <span>
        Арт
      </span>
    </a>
    <nav
      className="c3"
    >
      <ul>
        <li>
          <a
            href="/"
          >
            Главная
          </a>
        </li>
        <li>
          <a
            href="/works"
          >
            Работы
          </a>
        </li>
        <li>
          <a
            href="/reviews"
          >
            Отзывы
          </a>
        </li>
        <li>
          <a
            href="/contacts"
          >
            Контакты
          </a>
        </li>
      </ul>
    </nav>
  </div>
</header>
`;
